src/shogun/kernel/Kernel.h

```diff
@@ -860,6 +862,62 @@ class CKernel : public CSGObject
 		 * @return object casted to CKernel, NULL if not possible
 		 */
 		static CKernel* obtain_from_generic(CSGObject* kernel);
+
+		/** return auxiliary derivative with respect to left feature
+		 *
+		 * @param idx_l index of left feature (l_f)
+		 * @param idx_r index of right feature (r_f)
+		 *
+		 * @return gradient of a scalar auxiliary variable based on left feature and right feature wrt left feature
+		 *
+		 * This auxiliary variable is used in FITC inference for Gaussian processes
+		 * For example, in the GaussianKernel
+		 * \f[
+		 * k({\bf l_f},{\bf r_f})= exp(-\frac{||{\bf l_f}-{\bf r_f}||^2}{\tau})
+		 * \f]
+		 * where an auxiliary variable can be  
+		 * \f[
+		 *\frac{||{\bf l_f}-{\bf r_f}||^2}{\tau}
+		 * \f]
+		 */
+		virtual SGVector<float64_t> get_auxiliary_derivative_wrt_left_feature(int32_t idx_l, int32_t idx_r);
+
+		/** return auxiliary derivative with respect to left feature
+		 *
+		 * @param idx_l index of left feature (l_f)
+		 * @param idx_r index of right feature (r_f)
+		 *
+		 * @return value of a scalar auxiliary variable based on left feature and right feature
+		 * This auxiliary variable is used in FITC inference for Gaussian processes
+		 */
+		virtual float64_t get_auxiliary_value(int32_t idx_l, int32_t idx_r);
+
+		/** whether the kernel supports FITC inference
+		 *
+		 * @return false
+		 */
+		virtual bool supports_FITC() {return false;}
```

Comment:
I dont like the fact that the kernel base class contains GP related stuff.
Can we change this?